File: src/topic/src/demo04_sub_person.cpp
#include "ros/ros.h"
#include "topic/Person.h"


void doPerson(const topic::Person::ConstPtr &person){
    ROS_INFO("Subscrib message: %s, %d, %.2f",person->name.c_str(), person->age,person->height);

}


int main(int argc, char *argv[]){


    ros::init(argc, argv,"Subscribe");

    ros::NodeHandle nh;

    ros::Subscriber sub = nh.subscribe("chatter",10,doPerson);

    ros::spin();
    return 0;
}

File: src/topic/scripts/demo04_sub_person.py
import rospy
from topic.msg import Person


def callback(p):
    rospy.loginfo(f"Received person data: {p.name}, {p.age}, {p.height}")

def main():
    rospy.init_node("subscriber")
    sub = rospy.Subscriber("chatter", Person, callback)
    
    rospy.spin()

if __name__ == "__main__":
    main()